main_f4.c

```diff
@@ -331,6 +416,19 @@ main(void)
 	/* do board-specific initialisation */
 	board_init();
 
+	/* configure the clock for bootloader activity */
+	rcc_clock_setup_hse_3v3(&clock_setup);
+
+	/* start the interface */
+#if INTERFACE_USART
+	cinit(BOARD_INTERFACE_CONFIG_USART, USART);
```

Comment:
It depends on what you mean by reliable. If when the USART is initialised and the RX line is high and then it goes low for the break and then high again (as specified in the reference manual) then it is reliable. (But for it to be detected it has to have been high before AND after the break for the break flag to be set in the register).

Unfortunately it is just about impossible to have the line be high on startup and then pull it low for the break by the time the booloader sequence checks the RX pin. That is why I have put in the pin sampling stuff. But I see your point: if we basically never detect the break, but rather rely almost entirely on the sampling, then why not just do the sampling? I'll add this in shortly.